test/ambiguous.jl

```diff
@@ -1,406 +0,0 @@
-end
-
-## Other ways of accessing functions
-# Test that non-ambiguous cases work
-let io = IOBuffer()
-    @test precompile(ambig, (Int, Int))
-    cf = @eval @cfunction(ambig, Int, (Int, Int))
-    @test ccall(cf, Int, (Int, Int), 1, 2) == 4
-    @test length(code_lowered(ambig, (Int, Int))) == 1
-    @test length(code_typed(ambig, (Int, Int))) == 1
-end
-
-# Test that ambiguous cases fail appropriately
-let io = IOBuffer()
-    @test !precompile(ambig, (UInt8, Int))
-    cf = @eval @cfunction(ambig, Int, (UInt8, Int))  # test for a crash (doesn't throw an error)
-    @test_throws(MethodError(ambig, (UInt8(1), Int(2)), get_world_counter()),
-                 ccall(cf, Int, (UInt8, Int), 1, 2))
-    @test_throws(ErrorException("no unique matching method found for the specified argument types"),
-                 which(ambig, (UInt8, Int)))
-    @test length(code_typed(ambig, (UInt8, Int))) == 0
-end
-
-# Method overwriting doesn't destroy ambiguities
-@test_throws MethodError ambig(2, 0x03)
-ambig(x, y::Integer) = 3
-@test_throws MethodError ambig(2, 0x03)
-
-# Method overwriting by an ambiguity should also invalidate the method cache (#21963)
-ambig(x::Union{Char, Int8}) = 'r'
-@test ambig('c') == 'r'
-@test ambig(Int8(1)) == 'r'
-@test_throws MethodError ambig(Int16(1))
-ambig(x::Union{Char, Int16}) = 's'
-@test_throws MethodError ambig('c')
-@test ambig(Int8(1)) == 'r'
-@test ambig(Int16(1)) == 's'
-
-# Automatic detection of ambiguities
-
-const allowed_undefineds = Set([
-    GlobalRef(Base, :active_repl),
-    GlobalRef(Base, :active_repl_backend),
-    GlobalRef(Base.Filesystem, :JL_O_TEMPORARY),
-    GlobalRef(Base.Filesystem, :JL_O_SHORT_LIVED),
-    GlobalRef(Base.Filesystem, :JL_O_SEQUENTIAL),
-    GlobalRef(Base.Filesystem, :JL_O_RANDOM),
-])
-
-let Distributed = get(Base.loaded_modules,
-                      Base.PkgId(Base.UUID("8ba89e20-285c-5b6f-9357-94700520ee1b"), "Distributed"),
-                      nothing)
-    if Distributed !== nothing
-        push!(allowed_undefineds, GlobalRef(Distributed, :cluster_manager))
-    end
-end
-
-module Ambig1
-ambig(x, y) = 1
-ambig(x::Integer, y) = 2
-ambig(x, y::Integer) = 3
-end
-
-ambs = detect_ambiguities(Ambig1)
-@test length(ambs) == 1
-
-module Ambig2
-ambig(x, y) = 1
-ambig(x::Integer, y) = 2
-ambig(x, y::Integer) = 3
-ambig(x::Number, y) = 4
-end
-
-ambs = detect_ambiguities(Ambig2)
-@test length(ambs) == 2
-
-module Ambig3
-ambig(x, y) = 1
-ambig(x::Integer, y) = 2
-ambig(x, y::Integer) = 3
-ambig(x::Int, y::Int) = 4
-end
-
-ambs = detect_ambiguities(Ambig3)
-@test length(ambs) == 1
-
-module Ambig4
-ambig(x, y) = 1
-ambig(x::Int, y) = 2
-ambig(x, y::Int) = 3
-ambig(x::Int, y::Int) = 4
-end
-ambs = detect_ambiguities(Ambig4)
-@test length(ambs) == 0
-
-module Ambig5
-ambig(x::Int8, y) = 1
-ambig(x::Integer, y) = 2
-ambig(x, y::Int) = 3
-end
-
-ambs = detect_ambiguities(Ambig5)
-@test length(ambs) == 2
-
-
-using LinearAlgebra, SparseArrays, SuiteSparse
```

Comment:
TODO: Maybe parts of this test could be saved and not entirely discarded. But hard to tell which methods come from SuiteSpare (or which don't) without digging much deeper.

Reply:
Update: Running `make test-ambiguous` gives same results as master branch.
```
    JULIA test/ambiguous
Running parallel tests with:
  nworkers() = 1
  nthreads() = 1
  Sys.CPU_THREADS = 12
  Sys.total_memory() = 16.000 GiB
  Sys.free_memory() = 2.297 GiB

Test  (Worker) | Time (s) | GC (s) | GC % | Alloc (MB) | RSS (MB)
ambiguous  (1) |        started at 2022-09-17T12:27:10.240
ambiguous  (1) |     3.91 |   0.12 |  3.0 |     469.67 |   295.16

Test Summary: | Pass  Broken  Total  Time
  Overall     |  105       2    107  5.1s
    SUCCESS
```
after: `git checkout -f master && make clean && make test-ambiguous` master gives same results:
```
    JULIA test/ambiguous
Running parallel tests with:
  nworkers() = 1
  nthreads() = 1
  Sys.CPU_THREADS = 12
  Sys.total_memory() = 16.000 GiB
  Sys.free_memory() = 1.987 GiB

Test  (Worker) | Time (s) | GC (s) | GC % | Alloc (MB) | RSS (MB)
ambiguous  (1) |        started at 2022-09-17T12:50:23.021
ambiguous  (1) |     4.03 |   0.12 |  2.9 |     498.31 |   298.16

Test Summary: | Pass  Broken  Total  Time
  Overall     |  105       2    107  5.3s
    SUCCESS
```